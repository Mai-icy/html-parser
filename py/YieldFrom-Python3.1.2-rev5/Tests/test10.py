#
#   Test delegating 'throw'
#

def g1():
	try:
		print("Starting g1")
		yield "g1 ham"
		yield from g2()
		yield "g1 eggs"
	finally:
		print("Finishing g1")

def g2():
	try:
		print("Starting g2")
		yield "g2 spam"
		yield "g2 more spam"
	finally:
		print("Finishing g2")

g = g1()
for i in range(2):
	x = next(g)
	print("Yielded", x)
e = ValueError("tomato ejected")
g.throw(e)
