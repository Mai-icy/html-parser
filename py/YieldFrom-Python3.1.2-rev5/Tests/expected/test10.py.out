Starting g1
Yielded g1 ham
Starting g2
Yielded g2 spam
Finishing g2
Finishing g1
Traceback (most recent call last):
  File "test10.py", line 27, in <module>
    g.throw(e)
  File "test10.py", line 9, in g1
    yield from g2()
  File "test10.py", line 17, in g2
    yield "g2 spam"
ValueError: tomato ejected
